src/masternodeman.cpp

```diff
@@ -923,6 +919,10 @@ void ThreadCheckMasternodes()
 
             MilliSleep(1000);
             boost::this_thread::interruption_point();
+
+            // first clean up stale masternode payments data
+            masternodePayments.CleanPaymentList(mnodeman.CheckAndRemove(), mnodeman.GetBestHeight());
```

Comment:
isn't this an overkill?
it will get executed every second.

Reply:
guess that what you wanted to do is execute before the loop? Then should be moved outside of the while block.

Reply:
yeah good catch. Initially it was  inside  the `masternodeSync.IsBlockchainSynced()` and executed only with `c==1`.
But then thought it would have been better to clean the stale data before the sync.
Moving it outside of the while.

Reply:
updated